lnwallet/channel.go

```diff
@@ -547,6 +548,12 @@ type LightningChannel struct {
 	// way to lookup the original PaymentDescriptor.
 	rHashMap map[PaymentHash][]*PaymentDescriptor
 
+	// theirExpectedBalance reflects the available balance both locally
```

Comment:
I don't think this extra variable is required in order to enforce channel reserve semantics within the state machine itself. Instead, `validateCommitmentSanity` can be extended slightly to calculate the "expected balance" in real-time. This consolidates the commitment sanity checks (too many htlc's, reserve violation, etc) in once place, and will also still maintain its correctness once we move back to the sliding-window based commitment protocol.